app/server/logging/jsondb/src/main/java/io/syndesis/server/logging/jsondb/controller/PodLogMonitor.java

```diff
@@ -152,6 +159,37 @@ private void processLogStream(final InputStream is) throws IOException, Interrup
         }
     }
 
+    private static class InflightData {
+        Activity activity = new Activity();
+        Map<String, ActivityStep> steps = new HashMap<>();
+        Map<String, Object> metadata = new HashMap<>();
+
+        public ActivityStep getStep(String step, String id) throws IOException {
+            ActivityStep rc = steps.get(step);
+            if( rc == null ) {
+                rc = new ActivityStep();
+                rc.setId(id);
+                rc.setAt(KeyGenerator.getKeyTimeMillis(id));
+                steps.put(step, rc);
+            }
+            return rc;
+        }
+    }
+
+    InflightData getInflightData(String exchangeId, String logts) throws IOException {
+        InflightData data = inflightActivities.get(exchangeId);
+        if( data==null ) {
+            data = new InflightData();
+            data.activity.setPod(podName);
+            data.activity.setVer(deploymentVersion);
+            data.activity.setId(exchangeId);
+            data.activity.setAt(KeyGenerator.getKeyTimeMillis(exchangeId));
+            data.activity.setLogts(logts);
+            inflightActivities.put(exchangeId, data);
+        }
+        return data;
+    }
+
     @SuppressWarnings("PMD.CyclomaticComplexity")
     private void processLine(byte[] line) throws IOException {
```

Comment:
This is a really big method, couldn't we split it into smaller ones to make it a bit more explicit and easier to understand?